### PR TITLE
Integration tests for windows service support

### PR DESCRIPTION
**Pull request by guilhermocc**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
Running spire as a windows service inside windows containers.

**Description of change**
- Adds integration tests for windows service support.
- Fix windows service support not working when running spire in windows containers.

**Which issue this PR fixes**
Complement for #3490

### CODE REVIEW

File: pkg/common/entrypoint/entrypoint_windows.go [hunk 1]
```diff
@@ -20,7 +25,10 @@ type systemCall struct {
 }
 
 func (s *systemCall) IsWindowsService() (bool, error) {
-	return svc.IsWindowsService()
+	// We are using a custom function because the svc.IsWindowsService() one still has an open issue in which it states
+	// that it is not working properly in Windows containers: https://github.com/golang/go/issues/56335. Soon as we have
```
Comment (faisal-memon) [inline pkg/common/entrypoint/entrypoint_windows.go:29]: Is it worth submitting a PR.? That issue has a help wanted tag and it seems like you have a fix below.

Comment (guilhermocc) [reply pkg/common/entrypoint/entrypoint_windows.go:29]: A PR is already opened for this, in fact, the custom function with the workaround is based on the opened solution proposal: https://github.com/golang/sys/pull/141

File: test/integration/suites-windows/windows-service/00-setup.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+pwd
```
Comment (MarcosDY) [inline test/integration/suites-windows/windows-service/00-setup.sh:3]: why pwd is required?

File: test/integration/suites-windows/windows-service/01-start-server-service [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+source ./common
+
+docker-up spire-base
+
+create-service spire-server C:\\spire\\bin\\spire-server.exe
+start-service spire-server run -config C:/spire/conf/server/server.conf
```
Comment (MarcosDY) [inline test/integration/suites-windows/windows-service/01-start-server-service:7]: is it possible to be consistent and use `\\` or `/` in all places?

File: test/integration/suites-windows/windows-service/05-test-fetch-svid [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+log-debug "test fetch x509 SVID..."
+docker-compose exec -T -u ContainerUser spire-base  \
+    c:/spire/bin/spire-agent api fetch x509 || fail-now "failed to fetch x509"
+
+log-debug "test fetch JWT SVID..."
+docker-compose exec -T -u ContainerUser spire-base \
+    c:/spire/bin/spire-agent api fetch jwt -audience mydb fail-now "failed to fetch x509"
```
Comment (MarcosDY) [inline test/integration/suites-windows/windows-service/05-test-fetch-svid:9]:
```suggestion
docker-compose exec -T -u ContainerUser spire-base \
    c:/spire/bin/spire-agent api fetch jwt -audience mydb || fail-now "failed to fetch JWT"
```

File: test/integration/suites-windows/windows-service/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM spire-agent-windows:latest-local as spire-agent-windows
+
+FROM spire-server-windows:latest-local as spire-base
+
+COPY --from=spire-agent-windows C:/spire/bin/spire-agent.exe C:/spire/bin/spire-agent.exe
```
Comment (MarcosDY) [inline test/integration/suites-windows/windows-service/Dockerfile:5]: why you choose to use the same container to start both services?

File: test/integration/suites-windows/windows-service/00-setup.sh (hunk 2 above)
Comment (guilhermocc) [reply test/integration/suites-windows/windows-service/00-setup.sh:3]: Not sure if it is actually required, I took it from another test sample (windows workload attestor). I'm going to check if it is really necessary.

File: test/integration/suites-windows/windows-service/01-start-server-service (hunk 3 above)
Comment (guilhermocc) [reply test/integration/suites-windows/windows-service/01-start-server-service:7]: Sure! nice suggestion

File: test/integration/suites-windows/windows-service/Dockerfile (hunk 5 above)
Comment (guilhermocc) [reply test/integration/suites-windows/windows-service/Dockerfile:5]: So, in this initial version, I tried to reproduce a test that I ran manually on my local machine (spire server and agent on the same host). Still, I see that a more realistic scenario would be running them in different hosts, right? I can update it :)

Comment (MarcosDY) [reply test/integration/suites-windows/windows-service/Dockerfile:5]: yeah that is 'generally' the case

**Review by amartinezfayo — APPROVED** [review]
Thank you @guilhermocc, LGTM!

**Pull request merged**